realm/realm-library/src/main/java/io/realm/mongodb/sync/SubscriptionSet.java

```diff
@@ -0,0 +1,231 @@
+     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
+     * or wait for change listeners to trigger to see the downloaded data.
+     *
+     * If an error occurred, the underlying reason can be found through {@link #getErrorMessage()}.
+     *
+     * @param timeOut how long to wait for the synchronization to either succeed or fail.
+     * @param unit unit of time used for the timeout.
+     * @return {@code true} if all current subscriptions was accepted by the server and data has
+     * been downloaded, or {@code false} if an error occurred.
+     * @throws RuntimeException if the timeout is exceeded.
+     */
+    boolean waitForSynchronization(Long timeOut, TimeUnit unit);
+
+    /**
+     * Asynchronously wait for the subscription set to synchronize with the server. It will return when the
+     * server either accepts the set of queries and have downloaded data for them, or if an
+     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
+     * or wait for change listeners to trigger to see the downloaded data.
+     *
+     * If an error occurred, the underlying reason can be found through {@link #getErrorMessage()}.
+     *
+     * @param callback callback to trigger when the synchronization either succeed or fail. Results
+     *                 will be reported on the UI thread.
+     * @return {@code true} if all current subscriptions was accepted by the server and data has
+     * been downloaded, or {@code false} if an error occurred.
+     */
+    RealmAsyncTask waitForSynchronizationAsync(Callback callback);
+
+    /**
+     * Asynchronously wait for the subscription set to synchronize with the server. It will return when the
+     * server either accepts the set of queries and have downloaded data for them, or if an
+     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
+     * or wait for change listeners to trigger to see the downloaded data.
+     *
+     * If an error occurred, the underlying reason can be found through {@link #getErrorMessage()}.
+     *
+     * @param timeOut how long to wait for the synchronization to either succeed or fail.
+     * @param unit unit of time used for the timeout.
+     * @param callback callback to trigger when the synchronization either succeed or fail. Results
+     *                 will be reported on the UI thread.
+     * @return {@code true} if all current subscriptions was accepted by the server and data has
+     * been downloaded, or {@code false} if an error occurred.
+     */
+    RealmAsyncTask waitForSynchronizationAsync(Long timeOut, TimeUnit unit, Callback callback);
+
+    /**
+     * Modify the subscription set. If an exception is thrown during the update, no changes will be
+     * applied. If the update succeed, this subscription set is updated with the modified state.
```

Comment:
Also in multiple places:
```suggestion
     * applied. If the update succeeds, this subscription set is updated with the modified state.
```